### ISSUE TITLE
Gitpod prebuild is not running when PR created

### ISSUE THREAD

**Issue by shaal** [gitpod-io/gitpod#4925]
### Bug description

After configuring Gitpod Bot to run on all repositories
![image](https://user-images.githubusercontent.com/22901/126824799-9ebd110d-38f4-465f-ba1c-8c2790edb356.png)

I added the `prebuild` section to `.gitpod.yml`
https://github.com/shaal/DrupalPod/blob/main/.gitpod.yml#L61

I created a PR https://github.com/shaal/DrupalPod/pull/24

I see this comment in the PR
![image](https://user-images.githubusercontent.com/22901/126825087-df693d94-2e1f-4505-bf46-c4dc8bd25322.png)

Clicking on this button - open Gitpod workspace with that PR, but skipping the prebuild process (and I see the whole process running in terminal)

### Steps to reproduce

1. Find a repo that has init tasks
1. Configure Gitpod Bot to run on a repository
1. Make sure the `prebuild` section is in `.gitpod.yml`
1. Create a PR
1. Wait for the Bot to add "Open with Gitpod" button
1. Click on that button
1. Confirm Gitpod did not run prebuild and running Init tasks in terminal

Or

1. Create a new PR in https://github.com/shaal/DrupalPod/compare
1. Wait for the Bot to add "Open with Gitpod" button
1. Click on that button
1. Confirm Gitpod did not run prebuild and running Init tasks in terminal

### Expected behavior

1. A new PR should trigger Gitpod prebuild.
2. The button "Open in Gitpod" appear as a PR comment, only after the prebuild process has finished.

### Example repository

https://github.com/shaal/DrupalPod

**Comment by shaal** [comment]
Perhaps this is related?
While working on this branch https://github.com/shaal/DrupalPod/tree/setup-modes, before a PR is created -

When I push changes, and open in a new tab - https://gitpod.io/#https://github.com/shaal/DrupalPod/tree/setup-modes - 
I see the `prebuild` screen, which is what I expected to see. When the prebuild is done, the workspace is loaded with Gitpod interface, and I can see all the `init` commands running again in Terminal.

Only when I manually call for a prebuild of that branch - a "real" prebuild is run, and when the workspace opens, it is AFTER all `init` commands finished running, and the terminal only run `command` tasks.
